code/game/objects/structures/barricade/barricade.dm

```diff
@@ -466,3 +475,21 @@
 	nailgun.in_chamber = null
 	nailgun.load_into_chamber()
 	return TRUE
+
+// This proc is to check a bunch of condition to cancel the action that a welder user is trying to do while giving
+// a explanation on why...
+
+/obj/structure/barricade/proc/attackby_welder(obj/item/item, mob/user)
+	if(!HAS_TRAIT(item, TRAIT_TOOL_BLOWTORCH))
+		to_chat(user, SPAN_WARNING("You need a stronger blowtorch!"))
+		return FALSE
+
+	if(health == maxhealth)
+		to_chat(user, SPAN_WARNING("[src] doesn't need repairs."))
+		return FALSE
+
+	if(damage_state >= welder_fix && damage_state != null && welder_fix != null)
```

Comment:
This will runtime if either `damage_state` and `welder_fix` are `null`.

If you need to check a value is null you should do it before you try to compare it.
i.e.
```
if(isnull(value) || isnull(another_value))
    return
if(value > another_value)
    do_stuff()
```
If you want to specifically check for null use `isnull()`
```suggestion
	if(!(isnull(damage_state) || isnull(welder_fix)) && damage_state >= welder_fix)
```
This will return if either `damage_state` or `welder_fix` is null or both are not null and `damage_state >= welder_fix`

Reply:
if welder fix is null it mean barricade as no limit on the state in wish you can repair it.
so my check should return true if it's null and if damage state is null and damage_state>= welder fix...